### PR TITLE
Expose solver as a lib

### PR DESCRIPTION
**Pull request by JasonMarechal25**
Expose solver as a lib to allow external application (e.g. xpansion) to link to it and perform simulation directly without invoking external process in CLI.

- Move Application as a separate library. Scaffold for future work
- Expose solver-lib library. Existing solver executable now link to this library
- Refactor several solver libraries to fit proper library structure. Better and necessary to resolve some path.

### CODE REVIEW

File: src/solver/utils/include/antares/solver/utils/opt_period_string_generator.h [hunk 1]
```diff
@@ -17,6 +15,7 @@
    called.
 */
 
+#include "../../../../../optimisation/opt_period_string_generator_base.h"
```
Comment (flomnes) [inline src/solver/utils/include/antares/solver/utils/opt_period_string_generator.h:18]: Change this ?

Comment (JasonMarechal25) [reply src/solver/utils/include/antares/solver/utils/opt_period_string_generator.h:18]: later

**Comment by flomnes** [comment]
TODO 
Add generic entry points ?

**Comment by JasonMarechal25** [comment]
> TODO Add generic entry points ?

yes but at a later stage

File: src/solver/utils/include/antares/solver/utils/filename.h [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 #include <string>
 #include <memory>
-#include "../optimisation/opt_period_string_generator_base.h"
+#include "../../../../../optimisation/opt_period_string_generator_base.h"
```
Comment (a-zakir) [inline src/solver/utils/include/antares/solver/utils/filename.h:6]: wasn't possible without dots?

**Review by a-zakir — APPROVED** [review]

File: src/solver/utils/include/antares/solver/utils/filename.h (hunk 2 above)
Comment (a-zakir) [reply src/solver/utils/include/antares/solver/utils/filename.h:6]: you already answered!

**Review by flomnes — COMMENTED** [review]
Maybe you could add 1 or 2 tests and link them against solver-lib ? So as to have an example

**Pull request merged**